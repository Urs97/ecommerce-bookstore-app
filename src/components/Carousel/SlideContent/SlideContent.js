import React from 'react';
import './SlideContent.scss';
import { Link } from 'react-router-dom';

const SlideContent = ({text, title}) => {
    return (
        <section className="slide-content">
            <p>{text}</p>
            <h1 className="title">{title}</h1>
            <Link to='/store' className="link-button">See More</Link>
        </section>
    )
};

export default SlideContent;
