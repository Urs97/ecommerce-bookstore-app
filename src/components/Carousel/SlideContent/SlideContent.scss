@import '../../../styles/shared.scss';

.slide-content {
    position: absolute;
    z-index: 2;
    bottom: 0;
    left: $horizontal-content-padding;
    color: #ffffff;
    margin-bottom: 10em;
    font-size: 2.5rem;

    p {
        font-size: 1.5rem;
    }

    h1 {
        margin-bottom: 0.5em;
    }

    & .link-button {
        outline: none;
        text-decoration: none;
        color: #ffffff;
        font-size: 1.5rem;
        background-color: #000000;
        padding: 0.4em 0.6em;
        margin-top: 10em;
        border: 1px solid #ffffff;
        border-radius: 1em;
    }
    
    & .link-button:hover, & .link-button:focus {
        background-color: #202020;
    }
};