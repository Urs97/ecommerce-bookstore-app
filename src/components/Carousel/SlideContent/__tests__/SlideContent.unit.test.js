import React from 'react';
import {shallow} from 'enzyme';
import SlideContent from '../SlideContent';

describe('SlideContent', () => {
    test('renders empty SlideContent', () => {
        const wrapper = shallow(
            <SlideContent />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders SlideContent with props', () => {
        const mockText = "Reading provides you with wisdom."
        const mockTitle = "Keep Reading"
        const wrapper = shallow(
            <SlideContent 
                text={mockText} 
                title={mockTitle}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});