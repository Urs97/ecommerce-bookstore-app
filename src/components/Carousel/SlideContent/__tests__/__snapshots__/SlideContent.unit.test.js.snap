// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideContent renders SlideContent with props 1`] = `
<section
  className="slide-content"
>
  <p>
    Reading provides you with wisdom.
  </p>
  <h1
    className="title"
  >
    Keep Reading
  </h1>
  <Link
    className="link-button"
    to="/store"
  >
    See More
  </Link>
</section>
`;

exports[`SlideContent renders empty SlideContent 1`] = `
<section
  className="slide-content"
>
  <p />
  <h1
    className="title"
  />
  <Link
    className="link-button"
    to="/store"
  >
    See More
  </Link>
</section>
`;
